mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -520,6 +488,81 @@ class ALSSuite
   }
 }
 
+class ALSCleanerSuite extends SparkFunSuite {
+  test("Clean shuffles") {
+    val conf = new SparkConf()
+    val localDir = Utils.createTempDir()
+    val tempDir = Utils.createTempDir()
+    def getAllFiles: Set[File] =
+      FileUtils.listFiles(localDir, TrueFileFilter.INSTANCE, TrueFileFilter.INSTANCE).asScala.toSet
+    try {
+      conf.set("spark.local.dir", localDir.getAbsolutePath)
+      conf.set("spark.shuffle.manager", "sort")
+      val sc = new SparkContext("local[2]", "test", conf)
+      try {
+        sc.setCheckpointDir(tempDir.getAbsolutePath)
+        // Test checkpoint and clean parents
+        val filesBefore = getAllFiles
```

Comment:
Do we expect files generated automatically by `sc`? If so, we should leave a comment here and explain there are some initial files created by `sc.setCheckpointDir`. Otherwise, we don't need `filesBefore` because it should be empty.